### PR TITLE
Adds back purchasing the Autorifle and ammo through emagging the cargo supply console

### PR DESCRIPTION
**Pull request by CCC23**
This Is my first PR so be gentle if this doesn't work right. This is my attempt to align with the design document from https://hackmd.io/@6DbtsAKCTtW_9MByKFjZqg/r1xYKCNOt while keeping the weapon options for cargo uprisings more volatile and ballistics focused
## About The Pull Request

It adds back the Autorifle under the same way of unlocking the spec ops crate in cargo. Intended to give security a reason to bring out the bullet proof vests when dealing with a angry cargo tech instead of the normal laser based weaponry.
Right now a lot of our weapons are based on the laser charging style of ammo use with only three station available weapon types using actual bullet reloading these being the Mosin, Shotguns, and the detective revolver. Each of these weapons have their quirks but we are missing 1 crucial ballistic weapon. A reliable fast firing and namely not able to be shortened to normal size or smaller gun. That gun is the Autorifle. That's why I think adding the Autorifle back,if only in a fringe case is good for the game.
## Changelog

:cl:
add: Added New way to purchase the autorifle and ammo to cargo
balance: added more ballistics weapons to combat 
/:cl:

### CODE REVIEW

**Comment by CCC23** [comment]
shit gotta figure out why its doing that

**Comment by CCC23** [comment]
I'm draft this till I can find out what's wrong with it

**Comment by necromanceranne** [comment]
Well, the first problem is merging from master, but thats not necessarily going to cause a failure.

File: code/modules/cargo/packs/security.dm [hunk 1]
```diff
@@ -357,3 +357,26 @@
...
+	cost = CARGO_CRATE_VALUE * 7
+	hidden = TRUE
+	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
+                    /obj/item/ammo_box/magazine/wt550m9,
+                    /obj/item/gun/ballistic/automatic/wt550)
+    crate_name = "crate"
+	
+/datum/supply_pack/security/armory/wt550ammo
+    name = "Recalled Ammo Pack
+    desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. Each magazine is designed to facilitate rapid tactical reloads.\
```
Comment (necromanceranne) [inline code/modules/cargo/packs/security.dm:373]: I think your indentation here isn't working properly. You need to use tab indentation and not space indentation. I'm on my phone so I can't make a suggestion to fix it.

**Comment by Iamgoofball** [comment]
:-1:

it is trivial for the crew to get their hands on an emag, the second they kill exactly 1 traitor they're going to hand autorifles out to the entire crew

**Comment by Acer202** [comment]
I support this just on the basis that ballistics are more fun to use than energy weapons. More fun is good!

File: code/modules/cargo/packs/security.dm [hunk 2]
```diff
@@ -357,3 +357,26 @@
...
+	
+/datum/supply_pack/security/armory/wt550
+    name = "Recalled Weapon Pack"
+    desc = "Contains a set of old nanotranset brand autorifles recalled due to choking hazard."
+	cost = CARGO_CRATE_VALUE * 7
+	hidden = TRUE
+	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
+                    /obj/item/ammo_box/magazine/wt550m9,
+                    /obj/item/gun/ballistic/automatic/wt550)
+    crate_name = "crate"
```
Comment (Sealed101) [inline code/modules/cargo/packs/security.dm:369]:
```suggestion
	name = "Recalled Weapon Pack"
	desc = "Contains a set of old nanotranset brand autorifles recalled due to choking hazard."
	cost = CARGO_CRATE_VALUE * 7
	hidden = TRUE
	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
					/obj/item/ammo_box/magazine/wt550m9,
					/obj/item/gun/ballistic/automatic/wt550)
	crate_name = "crate"
```

Comment (Sealed101) [reply code/modules/cargo/packs/security.dm:369]: indents all over the place

File: code/modules/cargo/packs/security.dm [hunk 3]
```diff
@@ -357,3 +357,26 @@
...
+/datum/supply_pack/security/armory/wt550ammo
+    name = "Recalled Ammo Pack
+    desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. Each magazine is designed to facilitate rapid tactical reloads.\
+	Recalled due to Security demands."
+	cost = CARGO_CRATE_VALUE * 4
+	hidden = TRUE
+	contains = list(/obj/item/ammo_box/magazine/wt550m9,
+                    /obj/item/ammo_box/magazine/wt550m9,
+					)
+    crate_name = "crate"
```
Comment (Sealed101) [inline code/modules/cargo/packs/security.dm:380]:
```suggestion
/datum/supply_pack/security/armory/wt550ammo
	name = "Recalled Ammo Pack"
	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. Each magazine is designed to facilitate rapid tactical reloads.\
	Recalled due to Security demands."
	cost = CARGO_CRATE_VALUE * 4
	hidden = TRUE
	contains = list(/obj/item/ammo_box/magazine/wt550m9,
					/obj/item/ammo_box/magazine/wt550m9,
					)
	crate_name = "crate" 
```

Comment (Sealed101) [reply code/modules/cargo/packs/security.dm:380]: indents and a missing "

**Comment by Jackal-boop** [comment]
please please please please let the warden and HoS buy crates of these for extraordinary amounts i beg you

**Comment by CCC23** [comment]
> > 👎
> > it is trivial for the crew to get their hands on an emag, the second they kill exactly 1 traitor they're going to hand autorifles out to the entire crew
> 
> Security should detain and execute anyone that is carrying a ballistic weapon. Perhaps ballistics could punch holes through the station, causing breaches and encouraging security to take down shitters that use them.

A bit to strong and might lead to the gun being a disaster in the hands of erts. And a bit beyond the scope of my experience

**Comment by CCC23** [comment]
I think the main issue came from me trying to originally code this in notes

**Comment by CCC23** [comment]
Its fixed :)

File: code/modules/cargo/exports/weapons.dm [hunk 4]
```diff
@@ -74,3 +74,8 @@
 	unit_name = "pair"
 	message = "of handcuffs"
 	export_types = list(/obj/item/restraints/handcuffs)
+/datum/export/weapon/wt550
```
Comment (Fikou) [inline code/modules/cargo/exports/weapons.dm:77]:
```suggestion

/datum/export/weapon/wt550
```

File: code/modules/cargo/packs/security.dm [hunk 5]
```diff
@@ -357,3 +357,26 @@
...
 	crate_name = "thermal pistol crate"
+	
+/datum/supply_pack/security/armory/wt550
+	name = "Recalled Weapon Pack"
+	desc = "Contains a set of old nanotranset brand autorifles recalled due to choking hazard."
+	cost = CARGO_CRATE_VALUE * 7
+	hidden = TRUE
+	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/gun/ballistic/automatic/wt550)
```
Comment (Fikou) [inline code/modules/cargo/packs/security.dm:368]:
```suggestion
	contains = list(
		/obj/item/gun/ballistic/automatic/wt550 = 2,
		/obj/item/ammo_box/magazine/wt550m9 = 2,
	)
```

File: code/modules/cargo/packs/security.dm [hunk 6]
```diff
@@ -357,3 +357,26 @@
...
+	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. Each magazine is designed to facilitate rapid tactical reloads.\
+	Recalled due to Security demands."
+	cost = CARGO_CRATE_VALUE * 4
+	hidden = TRUE
+	contains = list(/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/ammo_box/magazine/wt550m9,
+					)
+	crate_name = "crate" 
+	
+
```
Comment (Fikou) [inline code/modules/cargo/packs/security.dm:382]:
```suggestion
	contains = list(/obj/item/ammo_box/magazine/wt550m9 = 2)
	crate_name = "crate" 

```

File: code/modules/cargo/packs/security.dm [hunk 7]
```diff
@@ -357,3 +357,26 @@
...
+	
+/datum/supply_pack/security/armory/wt550
+	name = "Recalled Weapon Pack"
+	desc = "Contains a set of old nanotranset brand autorifles recalled due to choking hazard."
+	cost = CARGO_CRATE_VALUE * 7
+	hidden = TRUE
+	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/gun/ballistic/automatic/wt550)
+	crate_name = "crate"
```
Comment (Fikou) [inline code/modules/cargo/packs/security.dm:369]: why

File: code/modules/cargo/packs/security.dm [hunk 8]
```diff
@@ -357,3 +357,26 @@
 	cost = CARGO_CRATE_VALUE * 7
 	contains = list(/obj/item/storage/belt/holster/thermal = 2)
 	crate_name = "thermal pistol crate"
+	
+/datum/supply_pack/security/armory/wt550
+	name = "Recalled Weapon Pack"
+	desc = "Contains a set of old nanotranset brand autorifles recalled due to choking hazard."
```
Comment (Fikou) [inline code/modules/cargo/packs/security.dm:363]:
```suggestion
	desc = "Contains a set of old Nanotrasen brand autorifles recalled due to choking hazard."
```

File: code/modules/cargo/packs/security.dm [hunk 9]
```diff
@@ -357,3 +357,26 @@
...
+	hidden = TRUE
+	contains = list(/obj/item/gun/ballistic/automatic/wt550,/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/ammo_box/magazine/wt550m9,
+					/obj/item/gun/ballistic/automatic/wt550)
+	crate_name = "crate"
+	
+/datum/supply_pack/security/armory/wt550ammo
+	name = "Recalled Ammo Pack"
+	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. Each magazine is designed to facilitate rapid tactical reloads.\
+	Recalled due to Security demands."
```
Comment (Fikou) [inline code/modules/cargo/packs/security.dm:374]:
```suggestion
	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. \
	Each magazine is designed to facilitate rapid tactical reloads. Recalled due to Security demands."
```

**Comment by GuillaumePrata** [comment]
> Security should detain and execute anyone that is carrying a ballistic weapon. Perhaps ballistics could punch holes through the station, causing breaches and encouraging security to take down shitters that use them.

We should bully Potato and see how viable it would be to have a "bullet hole" component or whatever else that bullets can apply to walls, windows and the floor.
Yes, pure excuse for more atmos content.

**Comment by CCC23** [comment]
Ok it was working and now I have to figure out which one broke it again

**Comment by CCC23** [comment]
ok I guess that works?

File: code/modules/cargo/packs/security.dm [hunk 10]
```diff
@@ -357,3 +357,26 @@
...
+	
+/datum/supply_pack/security/armory/wt550ammo
+	name = "Recalled Ammo Pack"
+	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. \
+	Each magazine is designed to facilitate rapid tactical reloads. Recalled due to Security demands."
+	cost = CARGO_CRATE_VALUE * 4
+	hidden = TRUE
+	contains = list(
+		/obj/item/ammo_box/magazine/wt550m9 = 2,
+		)
```
Comment (Mothblocks) [inline code/modules/cargo/packs/security.dm:380]:
```suggestion
	)
```

File: code/modules/cargo/packs/security.dm (hunk 7 above)
Comment (Mothblocks) [reply code/modules/cargo/packs/security.dm:370]: you marked this as resolved but did not answer why

File: code/modules/cargo/packs/security.dm [hunk 11]
```diff
@@ -357,3 +357,26 @@
...
+	hidden = TRUE
+	contains = list(
+		/obj/item/gun/ballistic/automatic/wt550 = 2,
+		/obj/item/ammo_box/magazine/wt550m9 = 2,
+	)
+	crate_name = "crate"
+	
+/datum/supply_pack/security/armory/wt550ammo
+	name = "Recalled Ammo Pack"
+	desc = "Contains four 20-round magazine for the Recalled WT-550 Auto Rifle. \
```
Comment (Mothblocks) [inline code/modules/cargo/packs/security.dm:374]: looks like it contains two. does each box have two?

Comment (CCC23) [reply code/modules/cargo/packs/security.dm:374]: Must have accidentally deleted one while debugging it

File: code/modules/cargo/packs/security.dm (hunk 7 above)
Comment (CCC23) [reply code/modules/cargo/packs/security.dm:370]: I don't know what they were asking in the why part but this has been changed to fit the style of other organization. In short one of fikous suggestions broke it so I had to figure out which one did

File: code/modules/cargo/packs/security.dm (hunk 11 above)
Comment (CCC23) [reply code/modules/cargo/packs/security.dm:374]: added the amount back to normal

File: code/modules/cargo/packs/security.dm (hunk 7 above)
Comment (Mothblocks) [reply code/modules/cargo/packs/security.dm:370]: They are asking why you are doing `crate_name = "crate"`

Comment (CCC23) [reply code/modules/cargo/packs/security.dm:370]: I think it was a relic part of me trying to get the crate to spawn correctly

Comment (CCC23) [reply code/modules/cargo/packs/security.dm]: that should fix it

**Comment by anclandor** [comment]
> 
> We should bully Potato and see how viable it would be to have a "bullet hole" component or whatever else that bullets can apply to walls, windows and the floor. Yes, pure excuse for more atmos content.

Me punching a whole though a meter thick bulkhead with a  single shot

**Comment by Jackal-boop** [comment]
> > We should bully Potato and see how viable it would be to have a "bullet hole" component or whatever else that bullets can apply to walls, windows and the floor. Yes, pure excuse for more atmos content.
> 
> Me punching a hole though a meter thick bulkhead with a single shot

Built different

**Comment by Mothblocks** [comment]
This is not the place for feature suggestions unrelated to the pull request itself, please take your conversation elsewhere

**Comment by CCC23** [comment]
autorifle :)

**Comment by necromanceranne** [comment]
Why the fuck are people talking about security balance when an antagonist needs to rock up and emag console for it to be an option.

I guess if Anne Tagonist dies with one of these in their pocket, sure, security gets autorifles, but that doesn't make them a default crew option. The same could be said of any antag gear if the crew gets their hands on it. This isn't related to the docs goals as a result, just reintroducing something taken out in a way that is more compliant.

That being said, what purpose _does_ this serve other than getting autorifles through cargo if a tot with an emag is looted. Tots dont really make the best use of autorifles, they're really more 'quantity over quality' weapons, and unless they hypnoflashed a mob of people (probably best use case), a lone antag doesn't stand to gain much. I'd focus on the actual antagonist this feature should stand to benefit and less on the impact on the round after the console is emagged, detached from the the antags own goals.

**Comment by CCC23** [comment]
> Why the fuck are people talking about security balance when an antagonist needs to rock up and emag console for it to be an option.
> 
> I guess if Anne Tagonist dies with one of these in their pocket, sure, security gets autorifles, but that doesn't make them a default crew option. The same could be said of any antag gear if the crew gets their hands on it. This isn't related to the docs goals as a result, just reintroducing something taken out in a way that is more compliant.
> 
> That being said, what purpose _does_ this serve other than getting autorifles through cargo if a tot with an emag is looted. Tots dont really make the best use of autorifles, they're really more 'quantity over quality' weapons, and unless they hypnoflashed a mob of people (probably best use case), a lone antag doesn't stand to gain much. I'd focus on the actual antagonist this feature should stand to benefit and less on the impact on the round after the console is emagged, detached from the the antags own goals.

The purpose of it originally is to make a hypnoflashed mob of people. I was trying to either get cargotechs to move towards the cargonia uprising style of traitor or to give a long arm ballistic weapon to use in shootout. Sure if some tider gets their hands on a emag they can now buy it. But at the end of the day that gun is still obtainable just with less ammo and less frequently.

**Comment by CCC23** [comment]
I keep crying for feedback but god wont answer my crys

**Comment by necromanceranne** [comment]
> The purpose of it originally is to make a hypnoflashed mob of people. I was trying to either get cargotechs to move towards the cargonia uprising style of traitor or to give a long arm ballistic weapon to use in shootout. Sure if some tider gets their hands on a emag they can now buy it. But at the end of the day that gun is still obtainable just with less ammo and less frequently.

I never replied but if thats the goal, sounds good and fine. Mothblocks already merged but yeah seems good.

**Pull request merged**